feature: Using pagination when calling API endpoints DOCS-245 (#677)
codacy/docs: docs/codacy-api/using-the-codacy-api.md

For example, to call the endpoint [searchRepositoryIssues](https://api.codacy.com/api/api-docs#searchrepositoryissues) specifying the issue levels `Error` and `Warning` in the body:

```bash
curl -X POST 'https://app.codacy.com/api/v3/analysis/organizations/gh/codacy/repositories/docs/issues/search' \
     -H 'api-token: SjE9y7ekgKdpaCofsAhd' \
     -H 'Content-Type: application/json' \
     -d '{"levels": ["Error", "Warning"]}'
```

## Using pagination

Endpoints that return lists containing a potential large number of results use cursor-based pagination to return the results in small batches:

> -   Calling endpoints that use pagination returns the first page of results together with a `pagination` object that includes a `cursor`.

Reviewer: I was very confused with the beginning of the sentence, I suggest something like this

```suggestion
-   These endpoints will return the results together with a `pagination` object that includes a `cursor`.
```

Reviewer: I would avoid saying "first page" because it is a condition for all the pages

Author: Good feedback, initially I had these explanations intermixed with the examples so they would tell a "story". But now it makes more sense to have this more generic. :+1:
